invenio_vocabularies/contrib/subjects/config.py

```diff
@@ -31,6 +31,7 @@ class SubjectsSearchOptions(SearchOptions):
         filter_field="scheme",
         fields=[  # suggest fields
             "title.*^100",
```

Comment:
minor: actually looking at this, it's going to probably suffer from the same issue with the "too many search clauses (max 1024)" when searching for more than 3-4 words.

We should apply the same style as in [e.g. affiliations](https://github.com/inveniosoftware/invenio-vocabularies/blob/f36aa0abf273e46c0e2255a1481e89ba21bd46fe/invenio_vocabularies/contrib/affiliations/config.py#L24-L41), so the search order should be:

1. Generic/common title field (in this case `subject`)
2. Acronyms (N/A for subjects)
3. Localized title (i.e. `title.{user_locale}`)
4. Synonyms/aliases (in this case `synonyms`)